heron/api/src/java/com/twitter/heron/api/windowing/triggers/AbstractBaseTriggerPolicy.java

```diff
@@ -0,0 +1,128 @@
+// Copyright 2017 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.twitter.heron.api.windowing.triggers;
+
+import java.io.Serializable;
+import java.util.Map;
+
+import com.twitter.heron.api.windowing.EvictionPolicy;
+import com.twitter.heron.api.windowing.TriggerHandler;
+import com.twitter.heron.api.windowing.TriggerPolicy;
+import com.twitter.heron.api.windowing.WindowManager;
+
+
+public abstract class AbstractBaseTriggerPolicy<T extends Serializable, S>
+    implements TriggerPolicy<T, S> {
+  protected TriggerHandler handler;
+  protected EvictionPolicy<T, ?> evictionPolicy;
+  protected WindowManager<T> windowManager;
+  protected Boolean started;
+  protected Map<String, Object> topoConf;
+
+  private boolean requiresEvictionPolicy = false;
+  private boolean requiresWindowManager = false;
+  private boolean requiresTopologyConfig = false;
+
+  /**
+  * Set the requirements in the constructor
+  */
+  public AbstractBaseTriggerPolicy(boolean requiresEvictionPolicy,
```

Comment:
Not sure if we need these checks.  I think it should be up to the user to make sure to pass in the correct objects he or she needs for a custom trigger.  Having to pass these booleans into the constructor also makes the interface confusing to some reading to code.  If you really want these checks in place, I would suggest using setters to set these.  Then it is clear to the reader of the code what is being set.

Reply:
@jerrypeng These checks were in place to keep the existing trigger policies as restrictive as they were before wrt which of the objects were attached.  I could provide an alternate no-arg constructor that sets them all to false, then use setters instead?  However, since this is the abstract class which already partially implements the TriggerPolicy interface, I would think it would be okay to perform some checks, but I have no issue with removing them either.

Reply:
lets remove them to keep the interface as simple and clean as possible

Reply:
@jerrypeng will do